File: common/src/main/java/net/onelitefeather/titan/common/blockhandler/JukeboxHandler.java
/**
 * Copyright 2025 OneLiteFeather Network
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.onelitefeather.titan.common.blockhandler;

import net.kyori.adventure.key.Key;
import net.minestom.server.instance.block.BlockHandler;
import org.jetbrains.annotations.NotNull;

public final class JukeboxHandler implements BlockHandler {

	private static final @NotNull Key NAMESPACE_ID = Key.key("minecraft:jukebox");

	@Override
	public @NotNull Key getKey() {
		return NAMESPACE_ID;
	}
}

File: src/main/kotlin/net/onelitefeather/titan/blockhandler/JukeboxHandler.kt
package net.onelitefeather.titan.blockhandler

import net.kyori.adventure.key.Key
import net.minestom.server.instance.block.BlockHandler
import net.minestom.server.tag.Tag
import net.minestom.server.utils.NamespaceID

class JukeboxHandler : BlockHandler {
    override fun getNamespaceId(): NamespaceID = NamespaceID.from(Key.key("minecraft:jukebox"))

    private val jukeTags = listOf(
        Tag.Boolean("IsPlaying"),
        Tag.ItemStack("RecordItem"),
        Tag.Long("RecordStartTick"),
        Tag.Long("TickCount"))

    override fun getBlockEntityTags(): MutableCollection<Tag<*>> {
        return jukeTags.toMutableList()
    }
}